src/test/java/teammates/test/cases/testdriver/BackDoorTest.java

```diff
@@ -266,22 +266,23 @@ public void testEditStudent() {
     @Test
     public void testCreateFeedbackResponse() {
 
-        FeedbackResponseAttributes fr = new FeedbackResponseAttributes();
         FeedbackQuestionAttributes fq = dataBundle.feedbackQuestions.get("qn1InSession1InCourse1");
         StudentAttributes student = dataBundle.students.get("student3InCourse1");
 
         fq = BackDoor.getFeedbackQuestion(fq.courseId, fq.feedbackSessionName, fq.questionNumber);
 
-        fr.feedbackSessionName = fq.feedbackSessionName;
-        fr.courseId = fq.courseId;
-        fr.feedbackQuestionId = fq.getId();
-        fr.feedbackQuestionType = fq.questionType;
-        fr.giver = student.email;
-        fr.giverSection = student.section;
-        fr.recipient = student.email;
-        fr.recipientSection = student.section;
-        fr.responseMetaData = new Text("Student 3 self feedback");
-        fr.setId(fq.getId() + "%" + fr.giver + "%" + fr.recipient);
+        FeedbackResponseAttributes fr = FeedbackResponseAttributes.builder()
+                .withFeedbackResponseId(fq.getId() + "%" + student.email + "%" + student.email)
+                .withFeedbackSessionName(fq.feedbackSessionName)
+                .withCourseId(fq.courseId)
+                .withFeedbackQuestionId(fq.getId())
+                .withFeedbackQuestionType(fq.questionType)
+                .withGiver(student.email)
+                .withGiverSection(student.section)
+                .withRecipient(student.email)
+                .withRecipientSection(student.section)
+                .withResponseMetaData(new Text("Student 3 self feedback"))
```

Comment:
Would it be better if the value here is more "dummy" instead of a magic value (that comes out of nowhere)? and it will be more consistent with your own created test as you have used `dummy#attribute`? I am fine with the current one as well just a thought.

Reply:
I guess it's to specify the kind of feedback response being tested here, since the test uses `student3InCourse1` and `recipientType` is `SELF`.

Reply:
Ok sure, I am fine with that.